data_steward/cdr_cleaner/cleaning_rules/update_invalid_zip_codes.py

```diff
@@ -0,0 +1,153 @@
+"""
+Sandbox and update invalid zip codes found in the observation table.
+
+Original Issues: DC-1633, DC-1645
+
+The intent of this cleaning rule is to sandbox and update any invalid zip code in the observation table. A zip code is
+    considered invalid if it does not match a zip3 code in the zip3 master lookup table or if it less than 5 digits.
+    The record is sandboxed and updated to have the following information:
+        value_as_string and value_source_value = 'Response removed due to invalid value'
+        value_as_number = 0
+        value_source_concept_id = 2000000010
+"""
+
+# Python imports
+import logging
+
+# Project imports
+from utils import pipeline_logging
+from constants.cdr_cleaner import clean_cdr as cdr_consts
+from common import JINJA_ENV, OBSERVATION, PIPELINE_TABLES, ZIP3_LOOKUP
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+
+LOGGER = logging.getLogger(__name__)
+
+# Creates sandbox that contains all invalid zip codes which are deemed invalid because they:
+# 1. Are not 5 digits in length
+# 2. Do not match a zip3 code in the master zip3 lookup table
+SANDBOX_INVALID_ZIP_CODES = JINJA_ENV.from_string("""
```

Comment:
I think there are 3 scenarios where this wouldn't give the desired results from the ticket specs:

1. In the case where the zip code has 5 digits, but the first 3 digits are not in the zip3_lookup table, the zip code will not be sandboxed as intended (e.g.  '00501'). The second LEFT JOIN in the query will not get a chance to run. Maybe replacing the joins with a UNION DISTINCT would work instead.

2. 
> # only use first 5 characters, and ensure they're only numeric in value

In the case where the zip code has 5 characters and the first 3 characters are a valid zip3 while the last two characters are nonnumeric, the record is not sandboxed as intended (e.g. '123ab').

3. 
> # get answer, trim value
I believe zip values should be trimmed of whitespace before other checks are run. Right now, a string like '    12345 ' would be sandboxed as invalid.